dynamiqs/time_tensor.py

```diff
@@ -239,3 +278,170 @@ def __add__(self, other: Tensor | TimeTensor) -> TimeTensor:
             return CallableTimeTensor(f, f0)
         else:
             return NotImplemented
+
+
+class PWCTimeTensor(TimeTensor):
+    # Arbitrary sum of PWC tensors.
+    #
+    # A single PWC tensor is defined by a tuple of 3 tensors (times, values, tensor),
+    # where
+    # - times: (nv+1) are the time points between which the PWC tensor take constant
+    #          values, where nv is the number of time intervals
+    # - values: (nv, ...) are the constant complex values for each time interval, where
+    #           (...) is an arbitrary batching size
+    # - tensor: (n, n) is the tensor value
```

Comment:
Two questions:

1. Shouldn't we allow formats `(n, m)`? I don't see any reason we should restrict to `(n, n)`, although this would be the main use.
2. Should we allow also the format `(nv, n, n)`? I can see applications where you would have a piece-wise constant tensor but the tensor itself changes from one piece to the next. We can still apply propagator solvers in this case.

Reply:
1. We _could_ actually allow arbitrary batching for the tensor, but I don't see why we _should_ as it's a tiny bit more complex.
2. Ongoing discussion on Slack.